### DISCUSSION TITLE
Side navigation component

### DISCUSSION THREAD

**Discussion by NemirStephan** [Legal-and-General/canopy#268] [Ideas] [answered]
I'd like to propose a secondary navigation component that could be passed an array of navigation items to display. The component could also have a content area to project in the navigated page (or router outlet)

**Comment by andy-polhill** [comment]
This is the big one 😱, but definitely high value. I think once we have a nice generic design in place, we should sketch out the proposed component API first, then work out the implementation that fits the API.

**Comment by NemirStephan** [comment, marked as answer]
So maybe not such a big one after all. As you have suggested @andy-polhill We can base a new component off the existing `lg-tab-nav-bar`, style the links differently and make it vertical. So pretty much a copy and paste of the lg-tab-nav-bar API

```html
<lg-side-nav-bar>
  <a lgSideNavBarLink id="side-nav-1" [isActive]="true" [routerLink]="">Nav 1</a>
  <a lgSideNavBarLink id="side-nav-2" [routerLink]="">Nav 2</a>
  <a lgSideNavBarLink id="side-nav-3" [routerLink]="">Nav 3</a>
</lg-side-nav-bar>
<lg-side-nav-content [selectedTabId]="side-nav-1">
  <p>Insert router-outlet here.</p>
</lg-side-nav-content>
```

#### lgSideNavBarLink Inputs
Name | Description | Type | Default | Required
-- | -- | -- | -- | --
index | Sets the nav index position in nav list | number | 0 | No
isActive | Sets the active state of the nav | boolean | false | No
isFocused | Sets the focus state of the nav | boolean | false | No

#### lgSideNavBarLink Outputs
Name | Description | Type | Default | Required
-- | -- | -- | -- | --
isActive | Sets the current active nav | boolean | undefined | No


#### lg-side-nav-content Inputs
Name | Description | Type | Default | Required
-- | -- | -- | -- | --
selectedNavId | Sets the id of the current selected nav | string | undefined | No

Designs to follow. But some things to note

- The content would appear above the nav on mobile and to the left of it on desktop
- There would be a margin margin between the nav and the content on mobile but none on desktop
- The styling of the nav and the links will be different on mobile and desktop
- We will need to add  additional content below the nav links (for a logout button for example). Can this be done simply by adding that content after the links?

**Reply by elenagarrone** [reply]
Do we have designs? I thought this was the work Fenton is currently investigating. 
How does it behave on mobile? We need to understand that first so that we can make our structure accessible.

**Reply by owensgit** [reply]
Update: side nav was created and ended up with a similar API:

https://github.com/Legal-and-General/canopy/tree/master/projects/canopy/src/lib/side-nav
